### PR TITLE
Show nullable types in operator QuickInfo tool tips

### LINKED ISSUE #21494: Better quickinfo for nullable operators
```C#
X509Certificate x = null;
if (x?.Subject?.Length >$$ 0)
{

}
```

QuickInfo on the `>` sign shows `int.operator >` when it should really show `int?.operator >`

image

### PR DESCRIPTION
**Pull request by Scott-Caldwell**
Prior to this PR, QuickInfo tool tips for operators on nullable types would only reference the underlying type instead of the nullable type. For example, the QuickInfo for the `>` in `a > b` where `a` and `b` are `int?` would read `bool int.operator >(int left, int right)`.

This affects compound assignments, increments/decrements, binary operators, and unary operators.

After this PR:
image
image
image
image

Fixes #21494.

### CODE REVIEW

**Comment by jcouv** [comment]
Tagging @AlekseyTs @gafter for thoughts on semantic API design.
In addition to lifted built-in operators, we should probably also consider lifted user-defined operators.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs [hunk 1]
```diff
@@ -7076,7 +7076,6 @@ ExpressionSyntax node4
 
             if (type.IsValueType && !type.IsPointerType())
             {
-                Assert.Equal(symbol1, symbol2);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs]: >Assert.Equal(symbol1, symbol2); [](start = 16, length = 31)

Asserts probably should not be removed, but changed to reflect the changes.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs [hunk 2]
```diff
@@ -8124,10 +8120,10 @@ struct TestStr
                         Assert.Equal("System.Boolean System.Object.op_Inequality(System.Object left, System.Object right)", info1.Symbol.ToTestDisplayString());
                         break;
                     case 8:
-                        Assert.Equal("System.Boolean System.Int32.op_Equality(System.Int32 left, System.Int32 right)", info1.Symbol.ToTestDisplayString());
+                        Assert.Equal("System.Boolean System.Int32?.op_Equality(System.Int32? left, System.Int32? right)", info1.Symbol.ToTestDisplayString());
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs]: >System.Int32? [](start = 53, length = 13)

Containing type change feels unexpected.

**Comment by AlekseyTs** [comment]
I am not sure why would we do this for intrinsic operators that aren't even real methods and wouldn't do this for real user-defined operators. At the same time I think that SemanticModel should give out symbols for real things, not the imaginary ones like lifted forms of operators etc.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs]: From IDE perspective, i would be ok with this.  @sharwell for thoughts as well.  

Note: it has been weird IMO that we show the underlying non-lifted operators.  Iknow i've been confused by this myself when using quick-info when working with nullable values and operators.  So i do appreciate improvements here.

**Comment by jcouv** [comment]
Feels like "synthesized" in 'SynthesizedIntrinsicOperatorSymbol' means that it's imaginary.  Otherwise, redundant with 'intrinsic'. 
Neal suggested we should have 'LiftedFrom' pointing back to a real operator.

**Comment by jaredpar** [comment]
Closing as this PR appears to have gone stale. Can re-open if submitter wants to pursue the fix.

**Pull request closed without merging**